File: src/Training.FileExplorer.ServerApp/Training.FileExplorer.Application/FileStorage/Models/Storage/StorageDirectory.cs
namespace Training.FileExplorer.Application.FileStorage.Models.Storage;

public class StorageDirectory : IStorageEntry
{
    public string Name { get; set; } = string.Empty;

    public string Path { get; set; } = string.Empty;

    public long ItemsCount { get; set; }

    public StorageEntryType EntryType { get; set; } = StorageEntryType.Directory;
}

File: src/Training.FileExplorer.ClientApp/src/infrastructure/models/entities/StorageDirectory.ts
import type { IStorageEntry } from "@/infrastructure/models/entities/IStorageEntry";
import { StorageEntryType } from "@/infrastructure/models/entities/StorageItemType";

export class StorageDirectory implements IStorageEntry {
    public name!: string;
    public path!: string;
    public itemsCount!: number;
    entryType: StorageEntryType = StorageEntryType.Directory;
}